huaweicloud/services/cci/resource_huaweicloud_cci_namespace.go

```diff
@@ -0,0 +1,370 @@
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"pool_size"},
+			},
+			"rbac_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCciNamespaceCreateParams(d *schema.ResourceData, config *config.Config) (namespaces.CreateOpts,
+	error) {
+	createOpts := namespaces.CreateOpts{
+		Kind:       "Namespace",
+		ApiVersion: "v1",
+	}
+	autoExpend := d.Get("auto_expend").(bool)
+	rbacEnable := d.Get("rbac_enable").(bool)
+	createOpts.Metadata = namespaces.Metadata{
+		Name: d.Get("name").(string),
+		Annotations: namespaces.Annotations{
+			Flavor:     d.Get("flavor").(string),
+			AutoExpend: strconv.FormatBool(autoExpend),
+		},
+		Labels: &namespaces.Labels{
+			EnterpriseProjectID: config.GetEnterpriseProjectID(d),
+			RbacEnable:          strconv.FormatBool(rbacEnable),
+		},
+	}
+	if size, isAdvance := d.GetOk("pool_size"); isAdvance {
+		poolSize := size.(int)
+		recyclingInterval := d.Get("recycling_interval").(int)
+		createOpts.Metadata.Annotations.PoolSize = strconv.Itoa(poolSize)
+		createOpts.Metadata.Annotations.RecyclingInterval = strconv.Itoa(recyclingInterval)
+		if enabled, ok := d.GetOk("container_network"); ok && enabled.(bool) {
+			createOpts.Metadata.Annotations.NetworkEnable = "vpc-network-ready"
+		}
+	}
+
+	return createOpts, nil
+}
+
+func resourceCciNamespaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.CciV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI v1 client: %s", err)
+	}
+
+	createOpts, err := buildCciNamespaceCreateParams(d, config)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to build createOpts of the CCI namespace: %s", err)
+	}
+	ns := d.Get("name").(string)
+	namespace, err := namespaces.Create(client, createOpts).Extract()
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI PVC: %s", err)
+	}
+	d.SetId(namespace.Metadata.UID)
+	stateRef := stateRefresh{
+		Pending:      []string{"Pending", "Active"},
+		Target:       []string{"Active"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        6 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+	if err := waitForCciNamespacestateRefresh(ctx, client, ns, stateRef); err != nil {
+		return err
+	}
+
+	return resourceCciNamespaceRead(ctx, d, meta)
+}
+
+func setCciNamespaceRbacEnable(d *schema.ResourceData, enabled string) error {
+	result, err := strconv.ParseBool(enabled)
+	if err != nil {
+		return err
+	}
+	return d.Set("rbac_enable", result)
+}
+
+func setCciNamespaceAutoExpend(d *schema.ResourceData, enabled string) error {
```

Comment:
maybe you can add type in json tag: Int64String int64 json:",string"

Reply:
made it